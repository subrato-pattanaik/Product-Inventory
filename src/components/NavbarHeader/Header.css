.navbrand {
  margin-left: 35px;
  margin-top: -18px;
}

.brandfont {
  font-size: 40px;
}

.nav-item .dropdown-menu {
  background: #343a40;
}

.nav-item .dropdown-item {
  color: #ffffff;
}

.nav-item .dropdown-item:hover,
.nav-item .dropdown-item:active {
  color: #ffc107;
  background: none;
}
.nav-item .dropdown-item:visited {
  background-color: #343a40 !important;
}
