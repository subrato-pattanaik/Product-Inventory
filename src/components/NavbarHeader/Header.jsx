import React, { useContext } from "react";
import { Navbar, Button } from "react-bootstrap";
import { withRouter, useHistory } from "react-router-dom";
import { UserContextAPI } from "../ContextAPI/UserContextAPI";
import "./Header.css";
import Navlinks from "./Navlinks";
function Header() {
  let navigation = useHistory();

  const { user, setUser } = useContext(UserContextAPI);
  return (
    <Navbar
      collapseOnSelect
      expand="lg"
      bg="dark"
      variant="dark"
      className="shadow"
    >
      <Navbar.Brand className="navbrand">
        <span className="brandfont">PI</span> Product Inventory
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Navlinks />
        {!!user ? (
          <>
            <Button
              variant="outline-warning"
              onClick={() => navigation.push("/Profile")}
            >
              Profile
            </Button>
            &emsp;
            <Button variant="outline-warning" onClick={() => setUser(false)}>
              Log out
            </Button>
          </>
        ) : (
          <>
            <Button
              variant="outline-warning"
              onClick={() => navigation.push("/Login")}
            >
              Log in
            </Button>
            &emsp;
            <Button
              variant="outline-warning"
              onClick={() => navigation.push("/Signup")}
            >
              Sign up
            </Button>
          </>
        )}
      </Navbar.Collapse>
    </Navbar>
  );
}
export default withRouter(Header);
