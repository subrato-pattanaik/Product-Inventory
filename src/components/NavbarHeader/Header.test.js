import React from "react";
import { shallow, mount } from "enzyme";

import Header from "./Header";
import { BrowserRouter } from "react-router-dom";

describe("Header Page Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Header.WrappedComponent />);
  });

  it("should test snapshot of Header page", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should test Navbar Component", () => {
    expect(wrapper.is("Navbar")).toEqual(true);
  });

  it("should test empty class component", () => {
    expect(wrapper.find("other-class").exists()).toEqual(false);
  });

  it("should test Nav bootstrap component which is inside Navlinks custom component", () => {
    const component = mount(
      <BrowserRouter>
        <Header.WrappedComponent />
      </BrowserRouter>
    );
    expect(component.find("Nav").exists()).toEqual(true);
  });
});
