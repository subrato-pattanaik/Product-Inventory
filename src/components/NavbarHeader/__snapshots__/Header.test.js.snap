// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Page Component should test snapshot of Header page 1`] = `
<Navbar
  bg="dark"
  className="shadow"
  collapseOnSelect={true}
  expand="lg"
  variant="dark"
>
  <NavbarBrand
    className="navbrand"
  >
    <span
      className="brandfont"
    >
      PI
    </span>
     Product Inventory
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Navlinks />
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="outline-warning"
    >
      Log in
    </Button>
    â€ƒ
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="outline-warning"
    >
      Sign up
    </Button>
  </NavbarCollapse>
</Navbar>
`;
